components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.management.apis/src/main/java/org/wso2/carbon/micro/integrator/management/apis/TemplateResource.java

```diff
@@ -0,0 +1,204 @@
+    private static final String SEQUENCE_TEMPLATE_TYPE = "sequence";
+
+    private org.apache.axis2.context.MessageContext axis2MessageContext;
+    private SynapseConfiguration synapseConfiguration;
+
+    public TemplateResource(String urlTemplate) {
+        super(urlTemplate);
+    }
+
+    @Override
+    public Set<String> getMethods() {
+        Set<String> methods = new HashSet<String>();
+        methods.add("GET");
+        methods.add("POST");
+        return methods;
+    }
+
+    @Override
+    public boolean invoke(MessageContext messageContext) {
+        axis2MessageContext = ((Axis2MessageContext) messageContext).getAxis2MessageContext();
+        synapseConfiguration = messageContext.getConfiguration();
+        if (Objects.isNull(axis2MessageContext) || Objects.isNull(synapseConfiguration)) {
+            return false;
+        }
+
+        String templateTypeParam = Utils.getQueryParameter(messageContext, TEMPLATE_TYPE_PARAM);
+        if (Objects.nonNull(templateTypeParam)) {
+            String templateNameParam = Utils.getQueryParameter(messageContext, TEMPLATE_NAME_PARAM);
+            if (Objects.nonNull(templateNameParam)) {
+                populateTemplateData(templateNameParam, templateTypeParam);
+            } else {
+                populateTemplateListByType(templateTypeParam);
+            }
+        } else {
+            populateFullTemplateList();
+        }
+        axis2MessageContext.removeProperty(Constants.NO_ENTITY_BODY);
+        return true;
+    }
+
+    /**
+     * Creates a response json with all templates available in the synapse configuration
+     * */
+    private void populateFullTemplateList() {
+        Map<String, Template> endpointTemplateMap = synapseConfiguration.getEndpointTemplates();
+        Map<String, TemplateMediator> sequenceTemplateMap = synapseConfiguration.getSequenceTemplates();
+        JSONObject jsonBody = new JSONObject();
+        JSONArray endpointTemplateList = new JSONArray();
+        JSONArray sequenceTemplateList = new JSONArray();
+        jsonBody.put(ENDPOINT_TEMPLATE_LIST, endpointTemplateList);
+        jsonBody.put(SEQUENCE_TEMPLATE_LIST, sequenceTemplateList);
+
+        if (Objects.nonNull(endpointTemplateMap) && !endpointTemplateMap.isEmpty()) {
+            for (Map.Entry<String, Template> templateEntry : endpointTemplateMap.entrySet()) {
+                Template endpointTemplate = templateEntry.getValue();
+                addToJsonList(jsonBody.getJSONArray(ENDPOINT_TEMPLATE_LIST), endpointTemplate.getName());
+            }
+        }
+        if (Objects.nonNull(sequenceTemplateMap) && !sequenceTemplateMap.isEmpty()) {
+            for (Map.Entry<String, TemplateMediator> templateEntry : sequenceTemplateMap.entrySet()) {
+                TemplateMediator sequenceTemplate = templateEntry.getValue();
+                addToJsonList(jsonBody.getJSONArray(SEQUENCE_TEMPLATE_LIST), sequenceTemplate.getName());
+            }
+        }
```

Comment:
We can use streams to improve readability.

```java
        endpointTemplateMap.forEach((key, value) -> {
            addToJsonList(jsonBody.getJSONArray(ENDPOINT_TEMPLATE_LIST), value.getName());
        });

        sequenceTemplateMap.forEach((key, value) -> {
            addToJsonList(jsonBody.getJSONArray(SEQUENCE_TEMPLATE_LIST), value.getName());
        });
```

Reply:
done